docs/src/testbeds.md

```diff
@@ -0,0 +1,88 @@
+# Testbeds
+
+Stylix provides a suite of virtual machines which can be used to test and
+preview themes without installing the target to your live system.
+
+These can be particularly helpful for:
+
+- Working on targets before the login screen, since you can avoid closing
+  your editor to see the result.
+- Developing for a different desktop environment than the one you normally use.
+- Reducing the risk of breaking your system while reviewing pull requests.
+
+Testbeds are also built by GitHub Actions for every pull request. This is less
+beneficial compared to running them yourself, since it cannot visually check
+the theme, however it can catch build failures which may have been missed
+otherwise.
+
+## Creation
+
+New testbeds are defined by creating a file called `testbed.nix` within the
+folder for the corresponding target. This file will automatically be loaded
+as a NixOS module, with options such as `stylix.image` already defined.
+The module should include any options necessary to install the target and
+any supporting software - for example, a window manager.
+
+If the target can only be used through Home Manager, you can write a
+Home Manager module within the NixOS module using the following format:
+
+```nix
+{
+    home-manager.sharedModules = [{
+        # Write Home Manager options here
+    }];
+}
+```
+
+Using `home-manager.sharedModules` is preferred over `home-manager.users.guest`
+since it allows us to easily change the username or add additional users in
+the future.
+
+Once the module is complete, use `git add` to track the file, then the new
+packages will be available to use with `nix run` as described below.
+
+## Usage
+
+You can list the available testbeds by running this command from anywhere
+within the repository:
+
+```sh
+nix flake show
+```
+
+Which prints something like:
+
+```
+github:danth/stylix
+└───packages
+    └───x86_64-linux
+        ├───docs: package 'stylix-book'
+        ├───palette-generator: package 'palette-generator'
+        ├───testbed-gnome-dark: package 'testbed-gnome-dark'
+        ├───testbed-gnome-light: package 'testbed-gnome-light'
+        ├───testbed-kde-dark: package 'testbed-kde-dark'
+        └───testbed-kde-light: package 'testbed-kde-light'
+```
+
+(This has been edited down to only the relevant parts.)
+
+The packages starting with `testbed-` are the testbeds.
```

Comment:
I intended this not to be updated and to just show the current list as an example.

Given that the docs are meant to be compiled with `nix build .#docs` and viewed through the website, we could consider replacing this entire section with an automatically generated table, and avoid directing users to `nix flake show` at all.

Advantages of that new method:
- Always up to date
- Will error if the format is changed, so you're forced to keep the docs in sync

Disadvantages:
- The raw markdown is less readable
- Additional code to maintain
- Only shows the current state of `master` unless you build the docs locally
- If the list becomes very long we may need to move it to a separate page, which breaks the flow of the text

Reply:
> Given that the docs are meant to be compiled with `nix build .#docs` and viewed through the website, we could consider replacing this entire section with an automatically generated table, and avoid directing users to `nix flake show` at all.

I am not sure whether the added documentation generation complexity is worth it.

Reply:
I agree, there are far more disadvantages than advantages on that list.

Note that the suggestion of `nix flake show | grep testbed` is a bit bugged since it outputs parts of the progress bar interleaved with the real output:

![screenshot](https://github.com/danth/stylix/assets/28959268/1d98c86e-129c-4e09-9d27-22dab016a7e1)

Whatever we need to do to work around this is probably too complex to include in the documentation without a full explanation, which distracts from the focus of the text.